library/bigip_ssl_certificate.py

```diff
@@ -0,0 +1,449 @@
+                return False
+            else:
+                raise F5ModuleError(e)
+        else:
+            return True
+
+    def exists_cert(self, name):
+        return self.api.tm.sys.crypto.certs.cert.exists(name=name)
+
+    def exists_key(self, name):
+        return self.api.tm.sys.crypto.keys.key.exists(name=name)
+
+    def import_cert(self, **kwargs):
+        name = kwargs['name']
+        cert_format = kwargs['cert_format']
+        if cert_format == 'pem':
+            try:
+                certfilename = kwargs['certfilename']
+            except KeyError:
+                certfilename = kwargs['name'] + '.crt'
+            else:
+                self.upload(certfilename)
+                self.install_cert(name, os.path.basename(certfilename))
+            try:
+                keyfilename = kwargs['keyfilename']
+            except KeyError:
+                pass
+            else:
+                self.upload(keyfilename)
+                self.install_key(name, os.path.basename(keyfilename))
+            return True
+        elif cert_format == 'pkcs12':
+            raise F5ModuleError("pkcs12 not supported by f5-sdk")
+
+    def install_cert(self, name, certfilename):
+        filename = os.path.join('/var/config/rest/downloads', certfilename)
+        params = {'from-local-file': filename, 'name': name}
+        self.api.tm.sys.crypto.certs.exec_cmd('install', **params)
+
+    def install_key(self, name, keyfilename):
+        filename = os.path.join('/var/config/rest/downloads', keyfilename)
+        params = {'from-local-file': filename, 'name': name}
+        self.api.tm.sys.crypto.keys.exec_cmd('install', **params)
+
+    def upload(self, filename):
+        self.api.shared.file_transfer.uploads.upload_file(filename)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            state=dict(type='str', default='present',
+                       choices=['absent', 'present']),
+            user=dict(type='str', required=True, aliases=['username']),
+            password=dict(type='str', required=True, no_log=True),
+            server=dict(type='str', required=True),
+            name=dict(type='str', required=True),
+            partition=dict(type='str', default='Common'),
+            connection=dict(type='str', default='rest',
+                            choices=['icontrol', 'rest'],
+                            aliases=['transport']),
+            cert_format=dict(type='str', default='pem',
+                             choices=['pem']),
+            cert_pem_file=dict(type='str'),
+            key_pem_file=dict(type='str'),
+            pkcs12_file=dict(type='str'),
+            pkcs12_password=dict(type='str', no_log=True),
+            validate_cert=dict(type='bool', default=True),
+            # overwrite=dict(type='bool', default=False)
+        )
+    )
+    result = {'changed': False}
+    if module.params['connection'] == 'icontrol':
```

Comment:
Lemme ask the REST framework developers about that deletion. I'm fine with this being a 12 or greater module. I'm treating these modules (generally) as an incentive to upgrade. So I'm deliberately being picky about providing SOAP support.

If there is an issue with the deleting thing, you can also try performing the operations inside of a Transaction.